commit b3d458bb87bc81ea340840a7707b5d06b9196fd2
Author: Agustin martin <agustin.martin@hispalinux.es>
Date:   Tue Sep 7 20:01:23 2010 +0200

    Make sure original ispell arg list is initialized in (ispell-start-process).
    
    * textmodes/ispell.el (ispell-start-process): Make sure original
      arg list is properly initialized (Bug#6993, Bug#6994).

diff --git a/lisp/textmodes/ispell.el b/lisp/textmodes/ispell.el
index e30da02..0e853cc 100644
--- a/lisp/textmodes/ispell.el
+++ b/lisp/textmodes/ispell.el
@@ -2591,12 +2591,13 @@ Keeps argument list for future ispell invocations for no async support."
                default-directory
              ;; Defend against bad `default-directory'.
              (expand-file-name "~/")))
+	 (orig-args (ispell-get-ispell-args))
          (args
           (append
-           (if (and ispell-current-dictionary ; Use specified dictionary.
-                    (not (member "-d" args))) ; Only define if not overridden.
+           (if (and ispell-current-dictionary      ; Not for default dict (nil)
+                    (not (member "-d" orig-args))) ; Only define if not overridden.
                (list "-d" ispell-current-dictionary))
-           (ispell-get-ispell-args)
+           orig-args
            (if ispell-current-personal-dictionary ; Use specified pers dict.
                (list "-p"
                      (expand-file-name ispell-current-personal-dictionary)))
